File: problems/hard/DNA_health/__tests__/test_dna_health.java
package problems.hard.DNA_health.__tests__;

import problems.hard.DNA_health.dna_health;

public class test_dna_health {
    public static void main(String[] args) {
        int n = 6;
        String[] genes = {"a", "b", "c", "aa", "d", "b"};
        int[] health = {1, 2, 3, 4, 5, 6};

        int[][] strands = {
            {1, 5},
            {0, 4},
            {2, 4}
        };
        String[] strandStrings = {"caaab", "xyz", "bcdybc"};

        int[] result = dna_health.computeDnaHealth(n, genes, health, strands, strandStrings);

        if (result[0] != 0 || result[1] != 19) {
            throw new AssertionError("❌ Test failed: Expected [0, 19], got [" + result[0] + ", " + result[1] + "]");
        }
        System.out.println("✅ All Java tests passed!");
    }
}

File: problems/hard/DNA_health/__tests__/test_dna_health.cpp
#include "../dna_health.h"
#include <iostream>
#include <cassert>

void test_dna_health() {
    int n = 6;
    vector<string> genes = {"a", "b", "c", "aa", "d", "b"};
    vector<int> health = {1, 2, 3, 4, 5, 6};
    vector<pair<int, int>> strands = {{1, 5}, {0, 4}, {2, 4}};
    vector<string> strandStrings = {"caaab", "xyz", "bcdybc"};

    auto result = computeDnaHealth(n, genes, health, strands, strandStrings);

    if (result.first != 0 || result.second != 19) {
        std::cout << "❌ Test failed: Expected [0, 19], got [" << result.first << ", " << result.second << "]\n";
    } else {
        std::cout << "✅ All C++ tests passed!\n";
    }
}

int main() {
    test_dna_health();
    return 0;
}
